src/main/java/com/artozersky/HackerNewsAPI/cache/impl/CacheEntityImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.artozersky.HackerNewsAPI.cache.impl;
+
+import com.artozersky.HackerNewsAPI.cache.CacheEntity;
+import com.artozersky.HackerNewsAPI.model.impl.NewsPostModelImpl;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+
+/**
+ * Implementation of the CacheEntity interface that provides a simple cache mechanism.
+ */
+public class CacheEntityImpl implements CacheEntity {
+
+    private final int maxSize;
+    private final Map<Long, NewsPostModelImpl> cache;
+
+    /**
+     * Constructs a CacheEntityImpl with a specified maximum size.
+     *
+     * @param maxSize the maximum number of entries the cache can hold
+     */
+    public CacheEntityImpl(int maxSize) {
+        this.maxSize = maxSize;
+        this.cache = new LinkedHashMap<Long, NewsPostModelImpl>(maxSize, 0.75f, true) {
+            @Override
+            protected boolean removeEldestEntry(Map.Entry<Long, NewsPostModelImpl> eldest) {
+                // Remove the eldest entry if the size of the map exceeds the maxSize
+                return size() > CacheEntityImpl.this.maxSize;
+            }
+        };
+    }
+
+    @Override
+    public NewsPostModelImpl get(Long key) {
+        return cache.get(key);
+    }
+
+    @Override
+    public List<NewsPostModelImpl> getAll() {
+        return new ArrayList<>(cache.values());
+    }
+
+    @Override
+    public void put(Long key, NewsPostModelImpl value) {
+        cache.put(key, value);
+    }
+
+    @Override
+    public void putAll(List<NewsPostModelImpl> allPosts) {
```

Comment:
those posts you put in the cache are already sorted by rank?

Reply:
yes. cacheQueue.offer(value); puts them in right order of priorityblockingqueue